src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Public.Data/DataSetVersionService.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces.Public.Data;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Model;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Model.Database;
+using Microsoft.EntityFrameworkCore;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services.Public.Data;
+
+public class DataSetVersionService(
+    ContentDbContext contentDbContext,
+    PublicDataDbContext publicDataDbContext) 
+    : IDataSetVersionService
+{
+    public async Task<List<DataSetVersionStatusSummary>> GetStatusesForReleaseVersion(Guid releaseVersionId)
+    {
+        var releaseFileIds = await contentDbContext
+            .ReleaseFiles
+            .AsNoTracking()
+            .Where(rf => rf.ReleaseVersionId == releaseVersionId && rf.File.Type == FileType.Data)
+            .Select(rf => rf.Id)
+            .ToListAsync();
+
+        return (await publicDataDbContext
+            .DataSetVersions
+            .AsNoTracking()
+            .Where(dataSetVersion => releaseFileIds.Contains(dataSetVersion.ReleaseFileId))
+            .Include(dataSetVersion => dataSetVersion.DataSet)
+            .ToListAsync())
+            .Select(dataSetVersion => new DataSetVersionStatusSummary(
+                Id: dataSetVersion.Id,
+                Title: dataSetVersion.DataSet.Title,
+                Status: dataSetVersion.Status)
+            )
+            .ToList();
```

Comment:
I would have thought you can get away without the `.Include` and the early `ToListAsync` here, moving it to the end and then no need for the additional parentheses?

Reply:
Unfortunately we can't, as the Select part is an expression tree and thus not usable in a Linq EF query.  This is why we have to do the actual query first and then the Select into the DataSetVersionStatusSummary afterwards.  If you've gotten examples of where we fetch directly into a DTO / ViewModel elsewhere though, feel free to point me at one and I'll see what we did there!  Thanks.

Reply:
I thought that's only the case if the expression tree cannot be translated to SQL by the provider?

In this case I can't see what can't be translated, so I would have thought this code should work?

```csharp
        return await publicDataDbContext
            .DataSetVersions
            .AsNoTracking()
            .Where(dataSetVersion => releaseFileIds.Contains(dataSetVersion.ReleaseFileId))
            .Select(dataSetVersion => new DataSetVersionStatusSummary(
                dataSetVersion.Id,
                dataSetVersion.DataSet.Title,
                dataSetVersion.Status)
            )
            .ToListAsync();
```

`RedirectsService#List`, 'MethodologyService#GetSummariesTree`, and `MethodologyService#GetPublishedPublicationsForMethodology` are some examples of where we are able to do this.

Reply:
Aha cool thanks.  It was trying to use named parameters that was confusing it.  Thanks for pointing out the examples.  I've switched it to not use explicit constructor param names now and it's happy.

Reply:
I've tried refreshing but I can't see that change 😆 .